### PR TITLE
Simplify build system for WASI

### PR DESCRIPTION
**Pull request by kateinoigakukun**
- Remove `if(NOT CMAKE_SYSTEM_NAME STREQUAL WASI)` as much as possible to be accepted for upstream. Use feature flag based building.
- Don't fetch libxml2 from the source and use CMake package system instead.
  - We need to define `LIBXML2_INCLUDE_DIR` and `LIBXML2_LIBRARY` when configuring.
  - These parameters will be given after https://github.com/swiftwasm/swift/pull/2733/files

- [Fix autolink issue for static library with @_implementationOnly](https://github.com/swiftwasm/swift/issues/2598). 
  - See also: https://forums.swift.org/t/autolinking-behavior-of-implementationonly-with-static-linking/44393
  - We still need to pass `-licui18n` because swift doesn't have a way to add public linking dependencies in module file. like `link` statement in modulemap.
    - I implemented this and after it will be merged, we can un-comment out [here](https://github.com/swiftwasm/swift-corelibs-foundation/pull/267/files#diff-9b52353c2832558b1c98bbe8ff81d4bba8ad404a1ab869661495b2ce444214e6R167-R171)

**This PR depends on https://github.com/swiftwasm/swift/pull/2733**

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
I checked this branch works well:  https://github.com/swiftwasm/swift/pull/2736/checks?check_run_id=1900758798

File: CoreFoundation/CMakeLists.txt [hunk 1]
```diff
@@ -555,17 +553,20 @@ if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
...
 install(TARGETS
-    CoreFoundation
-    CFXMLInterface
-    ${WASI_UNAVAILABLE_TARGETS}
-  DESTINATION
-    "${CMAKE_INSTALL_FULL_LIBDIR}/$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>")
+          CoreFoundation
+          CFXMLInterface
+        DESTINATION
+          # FIXME(katei): Install in WebAssembly.sdk which is isoalted from host sdk
```
Comment (MaxDesiatov) [inline CoreFoundation/CMakeLists.txt:560]:
```suggestion
          # FIXME(katei): Install in WebAssembly.sdk which is isolated from host sdk
```

File: CoreFoundation/CMakeLists.txt [hunk 2]
```diff
@@ -555,17 +553,20 @@ if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
...
+          "${CMAKE_INSTALL_FULL_LIBDIR}/$<LOWER_CASE:${CMAKE_SYSTEM_NAME}>")
+
+if(BUILD_NETWORKING)
+  install(TARGETS
+            CFURLSessionInterface
+          DESTINATION
+          # FIXME(katei): Install in WebAssembly.sdk which is isoalted from host sdk
```
Comment (MaxDesiatov) [inline CoreFoundation/CMakeLists.txt:567]:
```suggestion
          # FIXME(katei): Install in WebAssembly.sdk which is isolated from host sdk
```

File: Sources/Foundation/CMakeLists.txt [hunk 3]
```diff
@@ -163,6 +163,18 @@ set_target_properties(Foundation PROPERTIES
   Swift_MODULE_DIRECTORY ${CMAKE_BINARY_DIR}/swift
   INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_BINARY_DIR}/swift)
 
+if(NOT BUILD_SHARED_LIBS)
+  add_dependencies(Foundation CoreFoundation uuid)
+# TODO(katei): Comment out after swift-frontend implementation
```
Comment (MaxDesiatov) [inline Sources/Foundation/CMakeLists.txt:168]: Could you clarify what implementation are you referring to here?

Comment (kateinoigakukun) [reply Sources/Foundation/CMakeLists.txt:168]: added 👍

**Review by MaxDesiatov — APPROVED** [review]
Thanks!

**Pull request merged**